### DISCUSSION TITLE
simplyfing make_rng function in FLAX

### DISCUSSION THREAD

**Discussion by rabeehkarimimahabadi** [google/flax#2244] [Q&A] [answered]
Hi
In FLAX one need to use make_rng(name) on a set of pre-defined names, the issue is that if a user needs this inside one inner function, they dont have access to the name defined during .apply, so one needs to hard code this in the code, this design is complicated, comparing this with haiku one would do haiku.next_rng_key() which is very simple. I hope FLAX can get simplified on this at some point in future. Reading the documentation, I did not get this, and I hope the documentation for make_rng could get a bit more clear.

thank you!

**Comment by cgarciae** [comment, marked as answer]
Hey @rabeehkarimimahabadi! 

One of the main motivations for having named RNG groups vs no groups as in Haiku is that this lets you control how randomness behaves when using [Lifted Transformations](https://flax.readthedocs.io/en/latest/design_notes/lift.html) e.g. if you use [nn.vmap](https://flax.readthedocs.io/en/latest/_autosummary/flax.linen.vmap.html) over a module with more than one RNG groups you can choose which groups are "replicated" (same RNG values) and which are split (different RNG values).

I'll convert this to a discussion for now.
